### DISCUSSION TITLE
NextResponse 204 status

### DISCUSSION THREAD

**Discussion by wes337** [vercel/next.js#51118] [Help]
### Summary

How can I write the equivalent of:

```js
return res.status(204).end()
```

Using the next Nextjs 13 `NextResponse` object? I've tried:

```js
return NextResponse.json(null, { status: 204 });
```

and I get the error `TypeError: Response constructor: Invalid response status code 204`

**Comment by devjiwonchoi** [comment]
This feature is currently unavailable since `NextResponse` extends `Response` but did not implement the feature `status`.

![code](https://github.com/vercel/next.js/assets/120007119/50762871-95bc-41c4-90a9-53a90fd6fc39)

**Comment by lukeulrich** [comment]
You can send a 204 with `NextResponse` by constructing a new instance and setting the status in the second `init` parameter:

```ts
return new NextResponse(null, {status: 204});
```
